pkgs/applications/misc/asf/default.nix

```diff
@@ -0,0 +1,52 @@
+{ stdenv, fetchurl, unzip, makeWrapper, autoPatchelfHook
+, zlib, lttng-ust, curl, icu, openssl }:
+
+stdenv.mkDerivation rec {
+  pname = "ArchiSteamFarm";
+  version = "4.2.0.6";
+
+  src = {
+    x86_64-linux = fetchurl {
+      url = "https://github.com/JustArchiNET/ArchiSteamFarm/releases/download/${version}/ASF-linux-x64.zip";
+      sha256 = "05hx6q1lkbjbqhwi9xxvm7ycnsnpl1cnqzyy2yn0q4x27im399cn";
+    };
+    armv7l-linux = fetchurl {
+      url = "https://github.com/JustArchiNET/ArchiSteamFarm/releases/download/${version}/ASF-linux-arm.zip";
+      sha256 = "0l8irqrpl5vbjj84k4makj2ph2z6kpny7qz51zrzbgwhrlw0w4vg";
+    };
+    aarch64-linux = fetchurl {
+      url = "https://github.com/JustArchiNET/ArchiSteamFarm/releases/download/${version}/ASF-linux-arm64.zip";
+      sha256 = "0hg2g4i8sj3fxqfy4imz1iarby1d9f8dh59j266lbbdf2vfz2cml";
+    };
+  }.${stdenv.system} or (throw "Unsupported system: ${stdenv.system}");
+
+  nativeBuildInputs = [ unzip makeWrapper autoPatchelfHook ];
```

Comment:
Did you try our native framework?
https://github.com/NixOS/nixpkgs/blob/519519b6fcf0d23d1abd67ec43ec7548dc893cc0/doc/languages-frameworks/dotnet.section.md

Reply:
Upstream documentation on what is needed https://github.com/JustArchiNET/ArchiSteamFarm/wiki/Compatibility

Reply:
@veprbl That was a problem for me.
The docs you're refering are about development not packaing.
For packaging some trickery around nuget is needed (see e.g. pkgs/servers/nosql/eventstore or pkgs/development/tools/build-managers/msbuild).
I will look at how viable it is tomorrow.

Reply:
I thought you could use pre-built application and runtime from nixpkgs.

cc @jonringer

Reply:
@veprbl By looking at the link you kindly provided - generic package is not a way - it requires to use latest framework available. I've tried this, and there was an eror that dotnet package in nixpkgs is too old to run asf.
By wrapping dll from linux package that way it works, but I'm not sure will we always match the required .net version?

Reply:
Sorry, can't help you much with .net stuff. Also here is another useful link https://github.com/jb55/dotnet2nix/blob/master/dotnet-build.nix .

Reply:
@veprbl For the ease of maintenance I would prefer to go all binary until there is a decent dotnet packaging infrastructure. There was already a story where we went from sources to binary for electron/npm software exactly because of the lack of a proper infrastructure, it just started to give troubles: https://github.com/NixOS/nixpkgs/pull/76618

Reply:
@gnidorah I agree, it makes sense to stick with binaries for now.